Responsive layout

narcolepticinsomniac opened the pull request:
Hope I didn't forget anything.

I've been testing running the UI through devtools, which is how it came to my attention that our "responsive" layout falls apart much under 500px width:

![manage](https://user-images.githubusercontent.com/14100003/42174772-6296c02e-7df1-11e8-9aa4-993ae4403407.png)

![edit](https://user-images.githubusercontent.com/14100003/42174781-664bdf06-7df1-11e8-8ab9-2e996d8d58c2.png)


Aside from that, it really wasn't all that dynamic as far as wrapping content at normal widths. We were using `column-count` set to kick in at predefined media-screens, which is pretty rigid and is really calculated for the default English layout. With longer translated strings, wrapping at any widths is unpredictable, and should be done dynamically. 

I also reconfigured most margins so that they are at least partially created by bottom margins on the buttons themselves, so they aren't wrapping with no vertical margins between them.

I replaced editor page add/subtract "applies-to" buttons with plus/minus svg icons. Tophf mentioned a while back that we probably should, and he was right. They take up less space, so they remain well positioned and don't wrap till much narrower widths. We should really wrap the input with both buttons in a parent so they wrap together, but their function depends on their order so it'd need to be adjusted, and I was having no luck fixing it. That's really the only major responsive layout detail we should work out before it's good to go AFAICT. That function is in `edit/applies-to-line-widget.js` ~ line 43, and the input/buttons which need wrapping are in `edit.html` ~ line 103. Maybe now that I've narrowed the objective, @tophf will hook it up. 

As far as devtools integration, I'm gonna put that on the back-burner for now because there's a few too many limitations in the simplified embedded iframe implementation I was going for. It only works in Chromium, which kinda sucks, but not a deal-breaker for me. In Chromium, with some minor js adjustments, it communicates with the current tab, you can navigate through the whole UI, and almost all of the functionality works as expected. I really enjoy using it through devtools, but there's a couple big drawbacks. Mainly, communication with the current tab only works if devtools is docked, and warnings for exiting without saving don't work. Not sure if the latter could even be rectified.

Hopefully it's possible to properly tap into specific devtools API to communicate with the current tab in a more reliable way, because running the UI through devtools is a super convenient option IMO. I got the impression from the docs that communication is intentionally limited for security purposes, but there were workarounds by passing it through the background page. We'd still need to fix the unsaved edit alerts, which may or may not be possible using a proper API.

Either way, the simplified devtools integration is too limited to be worthwhile. It was easy to implement, so at least I didn't waste much time on it. It brought the responsive layout issues to my attention though, which did take some effort. All the interior wrapping is done with `flex` with next to no hard coded widths, so wrapping should be much more dynamic with unpredictable translated strings at any layout widths. Most of the rest of the responsive layout is also done with `flex`, except the outer manager, which needed to be compatible with the table for the style list. Should work the same as other pages though.

IIRC, the only absolute `max-width(s)` which might benefit from an adjustment are the manager filters, input and select. Some were just `flex-grow`, some also had a `max-width` calc for slightly under 100% to leave room for accompanying icons. The max-widths I used were determined by the hard-coded width limitation of the default header width, because at narrow UI widths, when the columns switch from two to one, the flex-grow allows for stupid looking widths. I wonder if the default header width limitation works decently with all translations to begin with, and also if the responsive max-widths could be improved.

My apathy kicks in when it comes to testing different translations, so hopefully some of our ESL users will test and provide feedback. @eight04 @DecentM @pabli @stonecrusher

Oh, I also added a very slight transition to `style-name` gradient hovers in the manager, and added the same effect in the popup. Just a suggestion to see if you guys like it.

narcolepticinsomniac commented:
Shit. Forgot to delete some leftover devtools code from `popup.css`.

eight04 commented:
I haven't see the code and just took some screenshots:

<details><summary>Usercss installation page</summary>

![2018-07-03 05 23 14](https://user-images.githubusercontent.com/1324510/42187275-07c21f7c-7e82-11e8-89b3-6b1d8dc9d0c6.png)
</details>

<details><summary>Usercss editor</summary>

![default](https://user-images.githubusercontent.com/1324510/42187123-8471ca28-7e81-11e8-9f92-79c84cec7877.png)
* There are three scrollbars on the page.
* The header is cropped which is inconsistent with the multi-section editor.
* Applies-to toolbar is broken that it doesn't show up.
</details>

<details><summary>Multi-section editor</summary>

![2018-07-03 05 28 11](https://user-images.githubusercontent.com/1324510/42187323-37c4401a-7e82-11e8-8542-0396c7a81cc7.png)
![2018-07-03 05 28 17](https://user-images.githubusercontent.com/1324510/42187324-37ee7a06-7e82-11e8-9898-fa06f78bd852.png)
![2018-07-03 05 28 26](https://user-images.githubusercontent.com/1324510/42187326-38179864-7e82-11e8-92a0-22ceed31d516.png)

* Looks good. I think using a "x" button instead of a "-" button would be more accurate. Sometimes +/- are also used in tree list/expander widgets. At the first glance I was thinking why do we have to make them collapsible individually.
</details>

<details><summary>Dashboard</summary>

![default](https://user-images.githubusercontent.com/1324510/42188087-0e534066-7e85-11e8-8bd5-ccd799fb5578.png)
* Horizontal scrollbar.

![2018-07-03 05 51 41](https://user-images.githubusercontent.com/1324510/42188124-3c2e3c98-7e85-11e8-8f1f-7ae48efcbc86.png)
![2018-07-03 05 52 49](https://user-images.githubusercontent.com/1324510/42188164-611e8d32-7e85-11e8-8e2a-bcfe7ac6b970.png)
![2018-07-03 05 54 19](https://user-images.githubusercontent.com/1324510/42188225-974ad654-7e85-11e8-8893-831bd9723d85.png)

* The header took the entire space (height 100%?)
* The layout after collapsing some sections doesn't look very well.

</details>

<details><summary>Multi-section editor (desktop)</summary>

![2018-07-03 05 56 13](https://user-images.githubusercontent.com/1324510/42188314-f98f2766-7e85-11e8-91e7-88db9250e4cf.png)

* The import/export button.

</details>

narcolepticinsomniac commented:
Damn. So you just get overflow everywhere? Admittedly, usercss editor pages didn't even occur to me, but that looks like all kinds of issues everywhere. Is it the same in English?

narcolepticinsomniac commented:
Yeah, single editor pages were overlooked, and need help. 

Multi-editor pages are OK? 

I can't repro overflow on usercss install pages or manager pages though. Text all looks like its wrapping, so what's causing overflow?

eight04 commented:
Probably related:
https://github.com/philipwalton/flexbugs/issues/39

narcolepticinsomniac commented:
Makes sense. I'll try to fix usercss editors and the FF overflow workaround sometime soon. Might be tomorrow though. Thanks for testing, and feel free to offer bug-fixes if you experiment in the meantime.

narcolepticinsomniac commented:
You'd think that two of the main browsers would be capable of at least displaying the same `display`, but FF requires a few extra properties here and there. I think it's all good now, can you confirm?.

FF also displays the fixed manager table a little differently. Looks like it's not quite as dynamic and causes overflow sooner, at slightly less narrow widths. We should probably just switch the installed style list to a slightly more compact `flex` to utilize the space better and more consistently cross-browser. That way we could get rid of the `table` display on the outer container too, and just make the whole manager page flex when the compact layout kicks in.

narcolepticinsomniac commented:
Every time I think everything is good, I have to recheck the other browser. I think they're both good now. Probably a couple redundant properties which could be deleted. I'll check again to tomorrow. 

As much feedback as anyone can provide from testing would be helpful.

narcolepticinsomniac commented:
>The import/export button.

I don't see an issue.

>The layout after collapsing some sections doesn't look very well.

This is how they were. Maybe they can be improved, but I've been trying to recreate the same layout with more responsive wrapping.

>Applies-to toolbar is broken that it doesn't show up.

That's a mystery to me. Not sure how I managed to break it, but it's not like it's positioned wrong, it's not even there. All I've done is rearranged the `display`. I haven't added any parents to change the order, so I'm not sure how I could've broken the `applies-to-line-widget.js`. 

We should also wrap these elements in a parent so they wrap together:
 
![applies-to](https://user-images.githubusercontent.com/14100003/42200869-3ac82214-7e63-11e8-8eb9-18021bf32002.png)

and fix the function in `applies-to-line-widget.js` for it as well, but the majority of that js is over my head.

tophf commented:
The code was using `button` tag selectors which is generally considered a bad idea. I'm looking at the code right now, gonna upload a fix here soon.

tophf commented:
As for the trouble with wrapping the applies-to items in a parent div, I don't see any need to alter js so here goes nothing.

tophf commented:
Is it fixed now?

narcolepticinsomniac commented:
Mostly all good. Add/remove all seems fixed. Only discrepancy is add buttons not being removed from usercss blocks. In the regular editor they're only on the last rule. Not sure if that matters.

tophf commented:
Uhm, weren't they always there? It's much more convenient to be able to add items anywhere. I think we need to add this feature to the standard editor.

narcolepticinsomniac commented:
Maybe in usercss? I certainly agree it's better. The regular editor should do the same.

narcolepticinsomniac commented:
Heh, they're hidden with our own CSS.

tophf commented:
I know. I'm fixing js that adds new items right now.

tophf commented:
Okay, now we need to make + and - buttons accessible via keyboard navigation like our other fancy buttons. IIRC we use `<a href="#">` wrapper and event.preventDefault() in the click listener or something.

narcolepticinsomniac commented:
Major upgrade. Thanks tophf.

tophf commented:
I see you've used tab characters instead of two spaces again.

narcolepticinsomniac commented:
Where at? Could've sworn I've avoided going full retard.

narcolepticinsomniac commented:
Were those it? Sublime is set to spaces, but it fucks off on the first line after opening curly brackets, which is pretty annoying. I wonder if there's some config for it.

tophf commented:
Use the find dialog and look for `\t` regexp.

narcolepticinsomniac commented:
Filter stats wrapping was an issue when I briefly checked Chinese. What a nightmare. Finding the setting to switch it back to English isn't so easy when all the settings are written in gibberish. I had to add a command line switch to change it back. It'd be a lot easier to check other translations if we could switch it in the extension.

Did I get all the tabs?

tophf commented:
Yep you got them.

Mottie commented:
When editing a non-usercss userstyle:
* I think the code block shouldn't have any right & left margins.
* I can't resize the editor height, so maybe we should make it full height? Then move the applies-to section to the top?

For usercss:
* The header appears fixed in place, and the editor scrolls underneath.
* If the editor is set to full height, this shouldn't happen. Also, the scroll bar is a dot.

General:
* The shortcuts popup should be full width, or at least wider. The right column is really compacted.
* The stylelint/csslint config popup should be wider.
* The old UI layout is still too wide. Should we detect for mobile platforms and auto-switch to the new UI?

tophf commented:
>the code block shouldn't have any right & left margins

Yes, in the narrow mode.

>I can't resize the editor height, so maybe we should make it full height? 

It's resizable here.

>Then move the applies-to section to the top?

Not sure we should suddenly change the order even if it'll match our usercss editor.

>The header appears fixed in place, and the editor scrolls underneath.
>If the editor is set to full height, this shouldn't happen. 

Yes, in the narrow mode.

>Also, the scroll bar is a dot.

Uhm, what?

>The shortcuts popup should be full width, or at least wider

The keymap shortcuts popup. Yes.

>The stylelint/csslint config popup should be wider.

I think we can just remove the line numbers column.

>The old UI layout is still too wide. Should we detect for mobile platforms and auto-switch to the new UI?

Which old UI? Screenshot please.
If you mean the style manager, I actually like the old UI on a phone.

Mottie commented:
I was using Chrome's "Toggle device toolbar" set to Pixel 2

![](https://user-images.githubusercontent.com/136959/42226865-b6180410-7ea5-11e8-8adc-241bb081804f.gif)

And what happens when I switch between new and old UI

![](https://user-images.githubusercontent.com/136959/42226881-c50eb874-7ea5-11e8-89bf-717b98e36fb0.gif)

eight04 commented:
I think buttons under "Mozilla Format" section are too large.
*master*
![default](https://user-images.githubusercontent.com/1324510/42236080-5e5bec3c-7f2c-11e8-9be0-37eb3827fca8.png)
*this PR*
![default](https://user-images.githubusercontent.com/1324510/42236095-691653e2-7f2c-11e8-8d32-d35c7cebd637.png)

stonecrusher commented:
<details><summary>edit.html in (small) desktop view</summary>
<img src="https://user-images.githubusercontent.com/1388389/42243586-25d6c86a-7f12-11e8-84f2-69e62aba74cc.png"/>

- I'd halve the right margin in desktop view as there is no shadow and therefore looks a bit too wide.

### Not responsive-specific:
- A different label for the beautify button under each section ("beautify section") would be good to distinguish it from the beautify button on the left which treats *all* sections.

- In the beautify popup an OK button (on the bottom left) would feel more natural than having to click the small x on top right. I already caught myself clicking the undo button by accident.

- The margin under the main heading ("Bereiche") is still pretty large.

- The plus sign for adding "applies to" is a bit sparse, but elegant. I think it would be better to have an empty "applies to domain" as default instead of global default. This forces the well known view with two forms and an additional minus sign next to the plus sign at the accustomed *right end*.
</details>
<br />
<details><summary>manage.html in mobile view</summary>
<img src="https://user-images.githubusercontent.com/1388389/42244816-b8ab2f92-7f16-11e8-83ab-d85ad5e4aa09.png"/>

- Why is there so much space till the stylelist starts?

- The column for the stylenames should have a percentual width which is smaller than current.

### Not responsive-specific:
- Shortcuts button is also misplaced here, should be deleted.

- The extra dropdown for favicons is quite ugly. The info about <code>https://www.google.com/s2/favicons</code> should be in a popup when clicking on info icon. The option to grey out favicons can be deleted at all if you ask me. That's something for styles. Or make it a separate option which only displays when favicons are activated.

</details>
<br />
<details><summary>Linter options</summary>
<img src="https://user-images.githubusercontent.com/1388389/42246081-908c9cd0-7f1b-11e8-963c-4d3110af085f.png"/>

### Not responsive-specific:
The text which says "show full list of rules for" at the bottom is the link and the following name of the linter which should be the link is missing completely.
</details>

tophf commented:
@stonecrusher, many of your observations/suggestions are outside of the scope of this PR. We'll handle them later, I guess. As for the last one, the English text doesn't have "for".

stonecrusher commented:
Yep, just thought it's a good opportunity if UI is changed anyway :-)
Link for list of rules has been changed 8 months ago and I missed it --> corrected in transifex german translation

narcolepticinsomniac commented:
>I think buttons under "Mozilla Format" section are too large.

Doesn't occur in English, so I had to force it to see the issue. Should be all set.


> I think the code block shouldn't have any right & left margins. (non-usercss editor)

Looks weird with none at all, but I reduced them a bunch.


>I can't resize the editor height, so maybe we should make it full height? Then move the applies-to section to the top? (non-usercss editor)

I have no issues resizing, although in the compact layout double-clicking `.resize-grip` seems not to expand enough.

Not a fan of moving applies-to either.


>I was using Chrome's "Toggle device toolbar" set to Pixel 2

So the scrollbar is not a dot?


>The header appears fixed in place, and the editor scrolls underneath. If the editor is set to full height, this shouldn't happen.

Well, unlike regular editor pages, the overflow in single-editor is handled by CM. No idea what you mean by "set to full height". AFAICT, the only sane layout there is what it is currently.


>`#help-popup` sizing

Should be good now.


>I think we can just remove the line numbers column.

I left the numbers and removed the CM gutter padding/functionality which makes them a more reasonable width. We could remove them altogether, but I think they're fine now.


>And what happens when I switch between new and old UI

`.applies-to` rules don't wrap, so I added `word-break: break-all` which fixes it. We could use ellipsis or maybe some of tophf's fancy hyphen js if you guys prefer.


The biggest remaining issue I see is the linter in compact mode. As you're editing, it pops up and moves the editor, then continues disappearing/reappearing and adding/removing errors, all of which make the editor bounce around while you try to use it. My only bright idea would be to always show the header/counter and make it click to open with no pref for open by default.

In fact, it'd probably be nice if all options collapsed by default in compact mode, although others should probably remember prefs. Not sure how complicated it'd be to have separate states/prefs for compact layouts.

Mottie commented:
Much better thanks!

> So the scrollbar is not a dot?

Yes, when looking at GitHub-dark - see the screen record I shared previously, I hover over the dot while scrolling. I found this [SO answer](https://stackoverflow.com/questions/26855012/make-a-codemirror-editor-and-its-gutter-100-height-of-container) which appears to work after refreshing the editor, but it leaves a big gap at the bottom.

> linter in compact mode

Maybe a fixed height on the `div` would work?

```css
#lint > div {
  height: 100px;
  overflow-y: auto;
}
```

silverwind commented:
> Sublime is set to spaces, but it fucks off on the first line after opening curly brackets, which is pretty annoying. I wonder if there's some config for it.

You should install [editorconfig-sublime](https://github.com/sindresorhus/editorconfig-sublime) to automatically set the editor to the correct indendation setting based on the current repo's `.editorconfig` file.

narcolepticinsomniac commented:
>You should install editorconfig-sublime 

Appreciate the tip, but I couldn't get it to fix this issue. I'm not all that experienced with configs for editors, but none I've found while googling work. Same shit in VS. Both say they're set to 2 spaces, but if you click enter after an opening curly bracket, they indent with a two space tab. It's fucking infuriating.

silverwind commented:
>  if you click enter after an opening curly bracket

Weird, I never experiences this. Are you using any autocomplete plugins which could be the culprit?

narcolepticinsomniac commented:
>Weird, I never experiences this.

It respects spaces unless you hit enter after an opening bracket. I don't get it, and I've wasted a ridiculous amount of time now on something so stupid. What're your configs if you don't mind? For Sublime in general and Edit Config?

narcolepticinsomniac commented:
Thanks @silverwind. Do Edit Config entries also go in the main Sublime config file? It has its own config, but no syntax seems to work in it. Anyway, I finally gave up and reinstalled from scratch, and it seems to have fixed it. Maybe I borked the install somehow.

@tophf How unrealistic would it be for options to collapse by default in compact mode? WDYT about the linter positioning issue in compact mode?

tophf commented:
I think that's a separate issue which we can handle later.

tophf commented:
My point is if it was that way before the PR then it's a separate issue.

narcolepticinsomniac commented:
Alright. Doesn't necessarily need to be fixed in this PR, but it shouldn't be left as it is. I imagine collapsing options might be related to fixing linter positioning if it's feasible. Otherwise, this PR is pretty solid, I think.

narcolepticinsomniac commented:
>`Preferences > Package Settings > EditorConfig`

I figured, but like I mentioned, the syntax is a mystery to me. I must've borked my Sublime install with plugins or configs. Reinstalling and adding some of your configs seems to have done the trick, so I don't need any global overrides for Edit Config either. Still would be nice to know the correct syntax for them. 

I tried the sample they provide on Github, and also attempted converting it to traditional JSON syntax, but couldn't get any of it to work. Don't need to experiment with it anymore, so it's not a priority. Thanks again.

narcolepticinsomniac commented:
Are we good with merging this PR?

tophf commented on install-usercss/install-usercss.css:
```diff
@@ -37,28 +38,21 @@ input:disabled + span {
   flex: 0 0 280px;
   box-sizing: border-box;
-  padding: 15px;
+  padding: 1rem;
```
We're using 15px in other files.

narcolepticinsomniac replied:
I assume from the CSS that it's a header replacement for warnings on invalid usercss code. Can anyone check if it's accounted for in the compact layout? Guess you'd need to force an error?

narcolepticinsomniac replied:
Checked out warnings. Should be good now.

tophf commented on edit/edit.css:
```diff
@@ -716,20 +774,27 @@ html:not(.usercss) .usercss-only,
 
 /************ reponsive layouts ************/
-@media(max-width: 850px) {
+@media(max-width: 800px) {
```
why was it changed?

narcolepticinsomniac replied:
Combined 2 or 3 screens into 1 in some cases. Best estimate of when compact becomes necessary, but we can switch back to 850px.

tophf replied:
My point is, if there was no problem with the existing number then there's no need to change it. Same for the other screens.

narcolepticinsomniac replied:
Sure, NBD.

tophf commented:
The right and left margins of the site column in the manager are too big now.

Old:

![1](https://user-images.githubusercontent.com/1310400/42443038-0c5812a6-8375-11e8-8a16-a0864712c794.png)

New: 

![2](https://user-images.githubusercontent.com/1310400/42443081-2decf0bc-8375-11e8-8a98-a7c5965ef2fb.png)

narcolepticinsomniac commented:
For this kinda update, we want as much confirmation that it's bug free in as many different languages as possible. Test it now, or forever hold your peace. The objective is better element wrapping in general, and far more responsive compact layouts at narrow widths, so either undock your browser and resize, or dock devtools to the side and resize by dragging their edge. Not really looking for subjective opinions on details, just bugs. A simple thumbs-up will do if everything is copacetic. 

@Mottie @eight04 @silverwind @DanaMW @stonecrusher @Hippyjake @DecentM @derv82 @JourneyOver

eight04 commented:
<details><summary>Dashboard</summary>

![default](https://user-images.githubusercontent.com/1324510/42548840-be5b4536-84fb-11e8-8207-203bf26f0977.png)
* `padding-bottom` becomes zero
</details>

<details><summary>Usercss editor</summary>

![default](https://user-images.githubusercontent.com/1324510/42548971-662445b0-84fc-11e8-9bfa-fb46d7f3b0d1.png)
* The inner padding of applies-to toolbar is not consistent.
* `+`/`-` buttons doesn't work. `event` is not defined.
  https://github.com/openstyles/stylus/blob/01606d35b047fdb7b01cfe6d3714aea46df81e36/edit/applies-to-line-widget.js#L44
  https://github.com/openstyles/stylus/blob/01606d35b047fdb7b01cfe6d3714aea46df81e36/edit/applies-to-line-widget.js#L77
</details>

<details><summary>Usercss installation page</summary>

![default](https://user-images.githubusercontent.com/1324510/42549391-778dcd6a-84fe-11e8-91d2-036f1fcf9bf4.png)
The editor is unusable when the header is large.
</details>

narcolepticinsomniac commented:
> Dashboard

Nothing to do with this PR. AFAICT, it's that way in the master, and only occurs in FF. Not to say we shouldn't fix it.

> The inner padding of applies-to toolbar is not consistent.

Simple fix.

>`+/-`  buttons doesn't work. event is not defined.

Yeah, works in the regular editor, not usercss apparently.

>Usercss installation page header

Those are pretty long strings at pretty narrow widths (and short height?). We could nowrap/ellipsis those long strings, or `max-height` the header with a scrollbar, or both. WDYT? Not like you really "use" the editor on install pages, but it shouldn't be hidden.

eight04 commented:
> Those are pretty long strings at pretty narrow widths (and short height?). We could nowrap/ellipsis those long strings, or max-height the header with a scrollbar, or both. WDYT? Not like you really "use" the editor on install pages, but it shouldn't be hidden.

How about giving the editor a min-height? So that users can scroll down to see the entire editor.

narcolepticinsomniac commented:
>So that users can scroll down to see the entire editor.

Unlike normal editor pages, usercss overflow is handled by CM. You wanna force double scrollbars at narrow widths? I don't use usercss, so maybe show me what you'd prefer.

eight04 commented:
I don't use mobile so I can't tell how double scrollbars act. If the installation page was designed for mobile, I would probably put the information and the source code in different tabs.

narcolepticinsomniac commented:
>see if my fixes broke anything.

Like I never broke it, thanks.

I guess the author is manipulating the meta to show a license badge on `openusercss.org`? If you install it from [Github](https://raw.githubusercontent.com/CephHunter/RS-wiki-DarkMode/master/Darkmode.user.css), the meta is correct. Syntax like that, where it doesn't belong, should either throw an error or be ignored.

![rs](https://user-images.githubusercontent.com/14100003/42728081-559ca7a6-8780-11e8-90cc-4a71e0c974aa.png)


>If the installation page was designed for mobile...

I'm not even testing mobile, so no, I'm not designing specifically _for_ it, but the layout being more responsive should make it much more usable. It'll work well on tablets, and at least be an improvement on smaller screens. The main benefit for the majority will probably be better wrapping in normal scenarios.


>I would probably put the information and the source code in different tabs.

I've suggested tabs for sectioned editors in the past, but I'm not sure they'd be good for usercss install pages. The editor is mostly there to take a quick glance at the code to check if it's as advertised. Not sure you'd wanna be clicking a tab before installing there.

That said, we should use nowrap/ellipsis and wrap the header content more compactly, so that's what I did. It might be nice to add something similar to the `.resize-grip` resizing to expand install editors to 100vh minus the title/install button. It'd also be good to auto-collapse all options everywhere in compact mode. Not necessarily in this PR, but both are pretty important IMO.

eight04 commented:
> I guess the author is manipulating the meta to show a license badge on openusercss.org?

OUCSS currently uses the markdown shown on the style homepage to generate `@description` so the original `@description` is overwritten.
https://github.com/OpenUserCSS/openusercss.org/issues/136

pabli24 commented:
- Polish language
- Edit style page

In left column options doesn't fit 
![](https://i.imgur.com/vPddxY6.png)
https://www.youtube.com/watch?v=ZeldQi0FFLc&feature=youtu.be
